### PR TITLE
Added solution for Project Euler problem 77.

### PR DESCRIPTION
**Pull request by fpringle**
Name: Prime summations

It is possible to write ten as the sum of primes in exactly five different ways:

7 + 3
5 + 5
5 + 3 + 2
3 + 3 + 2 + 2
2 + 2 + 2 + 2 + 2

What is the first value which can be written as the sum of primes in over five thousand different ways?

Reference: [https://projecteuler.net/problem=77](url)
Reference: #2695

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by dhruvmanila — COMMENTED** [review]
### Announcement:

@fpringle

*This message is sent to all pull requests which are **OPEN, NOT MERGED, not containing 'wontfix' label and contains 'Project Euler' or 'Project' or 'Euler' (case doesn't matter) in their title**. If this message doesn't apply to your pull request, please ignore this.*

### Message:

This is to notify all the users submitting a pull request for the Project Euler solution that the [README.md](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md) file has been updated with all the style requirements necessary for the directory and solution files. ***Please read through it and make all the necessary changes required to your submission.***

**Review by 53jk1 — APPROVED** [review]

File: project_euler/problem_077/sol1.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    """
+    Return the smallest integer that can be written as the sum of primes in over
+    m unique ways.
+    >>> solution(500)
+    45
+    """
+    for n in range(1, 100):
+        if len(partition(n)) > m:
+            return n
+    return 0
```
Comment (dhruvmanila) [inline project_euler/problem_077/sol1.py:62]: `0` becomes an answer as well. Maybe you want `None`?

File: project_euler/problem_077/sol1.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+"""
+
+primes = set(range(3, 100, 2))
+primes.add(2)
+for i in range(3, 100, 2):
+    if i not in primes:
+        continue
+    primes.difference_update(set(range(i * i, 100, i)))
+
+CACHE_PARTITION = {0: {1}}
```
[the remark this answers is not in the document]
Comment (fpringle) [reply project_euler/problem_077/sol1.py]: Thanks for the tip!

**Review by dhruvmanila — CHANGES_REQUESTED** [review]
Please provide descriptive parameter/variable names wherever the length is 1.

**Comment by fpringle** [comment]
> Please provide descriptive parameter/variable names wherever the length is 1.

Done

**Review by dhruvmanila — APPROVED** [review]
LGTM!

**Pull request merged**